src/fsharp/ExtensionTyping.fs

```diff
@@ -428,7 +428,7 @@ module internal ExtensionTyping =
         member __.GetGenericArguments() = x.GetGenericArguments() |> ProvidedType.CreateArray ctxt
         member __.ApplyStaticArguments(provider: ITypeProvider, fullTypePathAfterArguments, staticArgs: obj[]) = 
             provider.ApplyStaticArguments(x, fullTypePathAfterArguments,  staticArgs) |> ProvidedType.Create ctxt
-        member __.IsVoid = (typeof<System.Void>.Equals(x))
+        member __.IsVoid = (typeof<System.Void>.Equals(x) || (x.Namespace = "System" && x.Name = "Void"))
```

Comment:
Type.FullName returns "System.Void"

printfn "%s" (typeof<System.Void>.FullName);;
System.Void

Reply:
Yep, though that's ever so slightly slower....  Also FullName is causing hell here: https://github.com/fsprojects/FSharp.TypeProviders.SDK/issues/236#issuecomment-420251440 so preferring to rely on simpler primitives (under the hood the namespace and type name are stored separately)